sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -152,10 +159,7 @@ export class RequestResponseLink implements ReqResLink {
           request.to || "$management",
           context.message
         );
-        if (
-          request.message_id !== responseCorrelationId &&
-          request.correlation_id !== responseCorrelationId
```

Comment:
Service Bus doesn't set correlation_id on the requests. 
Python and .NET counterparts do not set or match for a correlation_id on the request. Hence, removing this check to be consistent.

Does removing this check affect event-hubs, @chradek ?

Reply:
So, previously when some of the requests did not have the message id set on them, would `responseCorrelationId` be undefined as well?

Reply:
> Does removing this check affect event-hubs, @chradek ?

correlation_id was never set on the request in Event Hubs either, so we should be good

Reply:
> So, previously when some of the requests did not have the message id set on them, would responseCorrelationId be undefined as well?

Yes!